Library/Homebrew/os/mac/sdk.rb

```diff
@@ -0,0 +1,57 @@
+require "os/mac/version"
+
+module OS
+  module Mac
+    class SDK
+      attr_reader :version, :path
+
+      def initialize version, path
+        @version = OS::Mac::Version.new version
+        @path = Pathname(path)
+      end
+    end
+
+    class SDKLocator
+      class NoSDKError < StandardError; end
+
+      def sdk_for v
+        cache_paths if @sdk_paths.nil?
+
+        path = @sdk_paths[v]
+        raise NoSDKError if path.nil?
+
+        SDK.new v, path
+      end
+
+      def latest_sdk
+        cache_paths if @sdk_paths.nil?
+        return if @sdk_paths.empty?
+
+        v, path = @sdk_paths.max
+        SDK.new v, path
+      end
+
+      private
+
+      def cache_paths
+        @sdk_paths = {}
```

Comment:
Also maybe make it always return the a hash (perhaps empty) meaning you can call it `sdk_paths` and skip the `nil` checks.